File: 序号/714.买卖股票的最佳时机含手续费.kt
import kotlin.math.*

class Solution {
    fun maxProfit(prices: IntArray, fee: Int): Int {
        var have = -prices[0]
        var notHave = 0
        for (i in 1 until prices.size){
            val tmpHave = have
            have = max(have,notHave-prices[i])
            notHave = max(notHave,tmpHave+prices[i]-fee)
        }
        return notHave
    }
}

File: 序号/714. 买卖股票的最佳时机含手续费.py
class Solution:
    def maxProfit(self, prices: List[int], fee: int) -> int:
        have,notHave=-inf,0
        for x in prices:
            have=max(have,notHave-x-fee)
            notHave=max(notHave,have+x)
        return notHave